File: CSharp/Kata/Solutions/MultiplesOf3Or5.cs
namespace Kata.Solutions;
//https://www.codewars.com/kata/514b92a657cdc65150000006/train/csharp
//https://www.codewars.com/users/MrAliSalehi/
public class MultiplesOf3Or5 : IKata
{
    /// <inheritdoc />
    public void Solution()
    {
        var answer = Solution(32);
    }

    public static int Solution(int value)
    {
        var allMultiples = new List<int>();

        for (var i = 0; i < value; i++)
        {
            if (IsMultipleOf3(i) || IsMultipleOf5(i))
                allMultiples.Add(i);
        }

        return allMultiples.Sum();
    }

    private static bool IsMultipleOf5(int value) => value % 5 == 0;
    private static bool IsMultipleOf3(int value) => value % 3 == 0;
}

File: FSharp/Kata/MultiplesOf3Or5.fs
//https://www.codewars.com/kata/514b92a657cdc65150000006/train/fsharp
//https://www.codewars.com/users/MrAliSalehi
let TotalSum(numbers : int32 list) =
    let mutable total = 0
    for num in numbers do
        total <- total + num;
    total

let solution x =
    let subNumbers = Seq.toList(seq{
        for i in 0..x - 1 ->
            if i % 3 = 0 || i % 5 = 0 then i
            else 0
    })
    TotalSum subNumbers


    
[<EntryPoint>]
let main argv = solution 10
